lusid-sdk-python/tests/test_transactions.py

```diff
@@ -0,0 +1,233 @@
+
+            dir_path = os.path.dirname(os.path.realpath(__file__))
+            with open(os.path.join(dir_path, "secrets.json"), "r") as secrets:
+                config = json.load(secrets)
+
+            token_url = os.getenv("FBN_TOKEN_URL", config["api"]["tokenUrl"])
+            username = os.getenv("FBN_USERNAME", config["api"]["username"])
+            password = pathname2url(os.getenv("FBN_PASSWORD", config["api"]["password"]))
+            client_id = pathname2url(os.getenv("FBN_CLIENT_ID", config["api"]["clientId"]))
+            client_secret = pathname2url(os.getenv("FBN_CLIENT_SECRET", config["api"]["clientSecret"]))
+            cls.api_url = os.getenv("FBN_LUSID_API_URL", config["api"]["apiUrl"])
+
+        else:
+            password = pathname2url(password_raw)
+            client_id = pathname2url(client_id_raw)
+            client_secret = pathname2url(client_secret_raw)
+
+        # Prepare our authentication request
+        token_request_body = ("grant_type=password&username={0}".format(username) +
+                              "&password={0}&scope=openid client groups".format(password) +
+                              "&client_id={0}&client_secret={1}".format(client_id, client_secret))
+        headers = {"Accept": "application/json", "Content-Type": "application/x-www-form-urlencoded"}
+
+        # Make our authentication request
+        okta_response = requests.post(token_url, data=token_request_body, headers=headers)
+
+        assert okta_response.status_code == 200
+
+        # Retrieve our api token from the authentication response
+        cls.api_token = {"access_token": okta_response.json()["access_token"]}
+
+        # Initialise our API client using our token so that we can include it in all future requests
+        credentials = BasicTokenAuthentication(cls.api_token)
+        cls.client = lusid.LUSIDAPI(credentials, cls.api_url)
+
+        # load the instruments using InstrumentLoader
+        cls.inst_loader = InstrumentLoader()
+        loader_response = cls.inst_loader.load_instruments(cls)
+
+        assert len(loader_response.values) == 5
+
+    def test_load_listed_instrument_transaction(self):
+
+        uuid_gen = uuid.uuid4()
+        scope = "finbourne"
+        effective_date = datetime(2018, 1, 1, tzinfo=pytz.utc)
+
+        # portfolio request
+        original_portfolio_id = "Id-" + str(uuid_gen)
+        request = models.CreateTransactionPortfolioRequest("Portfolio-" + str(uuid_gen),
+                                                           original_portfolio_id,
+                                                           base_currency="GBP",
+                                                           created=effective_date)
+        # create the portfolio
+        portfolio_response = self.client.create_portfolio(scope, request)
+
+        self.assertEqual(portfolio_response.id.code, request.code)
+
+        portfolio_id = portfolio_response.id.code
```

Comment:
You already have this as original_portfolio_id defined above, is there any reason to set it again off the response? We have checked that the response and the request have the same ID so we can be sure that it will be correct.

Reply:
cool. ive removed and now reuse portfolio_response.id.code